core/src/main/java/net/opentsdb/query/processor/groupby/GroupByTimeSeries.java

```diff
@@ -141,7 +141,7 @@ void addSource(final TimeSeries source, final int idx) {
     }
     // TODO - pool
     if (source_refs == null) {
-      source_refs = new int[16];
+      source_refs = new int[8];
```

Comment:
it starts with initial size of 8 and then keeps growing?

Reply:
Yup, reducing it for the case wherein we have 1 series per group. Still need to pool it eventually.